packages/utils-json/src/final-config.ts

```
@@ -31,16 +41,23 @@ export const finalConfig = <T extends ExtendableConfiguration> (config: T, resou
         const lastPath = configPath;
         const configDir = path.dirname(configPath);
 
-        configPath = path.resolve(configDir, finalConfigJSON.extends);
+        try {
+            configPath = require.resolve(finalConfigJSON.extends, { paths: [configDir] });
+        } catch (error) {
+            const castedError = error as IParsingError;
 
-        if (configIncludes.includes(configPath)) {
+            if (castedError && castedError.code === 'MODULE_NOT_FOUND') {
+                return getParsingError('Parent configuration missing', resource, error, 'MODULE_NOT_FOUND');
+            }
 
-            const error = new Error(`Circular reference found in file ${lastPath}`) as IParsingError;
-            const originalPathUri = getAsUri(configIncludes[0]);
+            throw error;
```

Is it intentional that this particular case throws instead of returns?